api/preprints/serializers.py

```diff
@@ -75,184 +83,105 @@ class PreprintSerializer(JSONAPISerializer):
         }
     )
 
-    contributors = RelationshipField(
-        related_view='nodes:node-contributors',
-        related_view_kwargs={'node_id': '<pk>'},
-        related_meta={'count': 'get_contrib_count'},
-    )
-
     class Meta:
         type_ = 'preprints'
 
     def get_preprint_url(self, obj):
-        kwargs = {
-            'node_id': obj._id,
-            'version': self.context['request'].parser_context['kwargs']['version']
-        }
-        return absolute_reverse('preprints:preprint-detail', kwargs=kwargs)
+        return absolute_reverse('preprints:preprint-detail', kwargs={'preprint_id': obj._id, 'version': self.context['request'].parser_context['kwargs']['version']})
 
     def get_absolute_url(self, obj):
         return self.get_preprint_url(obj)
 
     def get_doi_url(self, obj):
-        return 'https://dx.doi.org/{}'.format(obj.preprint_doi) if obj.preprint_doi else None
+        return 'https://dx.doi.org/{}'.format(obj.preprint_article_doi) if obj.preprint_article_doi else None
 
-    def create(self, validated_data):
-        node = Node.load(validated_data.pop('_id', None))
-        if not node:
-            raise exceptions.NotFound('Unable to find Node with specified id.')
+    def update(self, preprint, validated_data):
+        from website.models import Node
+        assert isinstance(preprint, PreprintService), 'You must specify a valid preprint to be updated'
```

Comment:
`nose.tools`-style asserts (`assert_equal`, etc.) won't work in production because `nose` isn't installed. Normal `assert` statements will because it's a python built-in. From [stackoverflow -- Best practice for Python Assert](http://stackoverflow.com/questions/944592/best-practice-for-python-assert#945135):

>Asserts should be used to test conditions _that should never happen_. The purpose is to crash early in the case of a corrupt program state.

That being said, `PreprintMixin` should catch this early if it's not a `PreprintService` object, and the call to `check_object_permissions` in it should do so for `User`/`Node` permissions. This is just an extra verification, and could be removed.